### PR TITLE
Add guidance on appropriate choice of version number for new plugins

### LINKED ISSUE #727: Guidance on appropriate choice of version number for new plugins
In pull request #725, a discussion came up around the instructions (on the IDEA page) about what to use when the IDE prompts you for a version number for your newly created project.  Currently the instructional text in the Sponge docs just says it is up to you.

If there is a standard for what kind of version numbering scheme we want Sponge plugins to use, it would be good to clarify this here (though maybe it should be a page in its own because it is not an issue specific to IDEA and one will have to decide on a version number regardless of the choice of IDE our abstinence from one.

I don't personally know what the right answer is.  I am aware of "semantic versioning" which is something like:

 - change the first number when you make a breaking change to the API
- change the middle number when you add new functionality to the API.
- change the last number for any other minor changes

This might not totally fit with plugins which don't have an externally facing API

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes #727 

If I forgot an important versioning scheme please tell me, don't light the 🔥 on your pitchforks. ^^

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/5a1bb94/.

Here are some links to the pages that were modified:

- added: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/5a1bb94/plugin/project/version-numbers.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/5a1bb94/plugin/project/gradle.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/5a1bb94/plugin/project/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/5a1bb94/plugin/project/maven.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/5a1bb94/plugin/workspace/eclipse.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/5a1bb94/plugin/workspace/idea.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/750#issuecomment-415871249), not to the direct url to the preview._

File: source/plugin/project/version-numbers.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+.. note::
+
+    The SpongeAPI uses this versioning scheme.
+
+**Examples**
+
+* 0.1.0
+* 1.0.0
+* 1.1.0-SNAPHOT
+* 3.12.21-MC1.12.2
```
Comment (phit) [inline source/plugin/project/version-numbers.rst]: maybe switch out with API7, sponge plugins really shouldnt use mc version but api version instead

File: source/plugin/project/version-numbers.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
+Version Numbers
+===============
+
+Your plugin needs a unique version number in order to allow humans to find out which version of a plugin is the
+successor of the other. There are plenty of versioning schemes that you could use for your own plugin. In the following
+part we explain a few concepts with their pros and cons.
+
+Sponge recommends to use a version identifier that consists of either two or three numeric version parts separated with
+a ``.`` and additional label appended with a hyphens ``-`` such as ``3.1`` or ``1.0.3-MC1.12.2`` similar to SemVer.
```
Comment (phit) [inline source/plugin/project/version-numbers.rst]: same as below API > MC version

File: source/plugin/project/version-numbers.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+* A change in the **MAJOR** version indicates that you made incompatible API changes. Rewriting large parts of your
+  plugin or the entire plugin would also count as incompatible changes.
+* A change in **MINOR** version indicates that a feature has been added without breaking backwards
+  compatibility.
+* A change in the **PATCH** version indicates that some bugs have been fixed.
+
+The labels can be used to mark builds as pre-releases or contain build metadata such as the supported minecraft version.
+If you increase one part of the version, then you will set the following parts back to zero. Using this strategy you
+can find the latest version by comparing the version blocks with each other, however this scheme does not contain any
```
Comment (Inscrutable) [inline source/plugin/project/version-numbers.rst:32]: It's a bit unclear what the part after "however" means. An example might help.

File: source/plugin/project/version-numbers.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+* Mostly API oriented so the terms don't really apply if you don't provide an API in your plugin.
+
+Timestamp
+---------
+
+This is a less commonly used versioning scheme, but it is also a very simple one, as it does not contain any additional
+information in the version numbers. It does not matter which separators you use (if any) or which date and time parts
+you include. However we recommend you to order date and time parts in the descending order of duration like
+``YYYY-MM-DD``. Please make sure that, if you sort your files alphabetically that the order is the same as the version
```
Comment (Inscrutable) [inline source/plugin/project/version-numbers.rst]: Comma in a bad place. I suggest "If you sort your files alphabetically, please make sure that the order is ..."

**Review by Inscrutable — COMMENTED** [review]
Grammatically sufficient.

File: source/plugin/project/version-numbers.rst (hunk 4 above)
Comment (ST-DDT) [reply source/plugin/project/version-numbers.rst]: I'm not sure whether it still contains the same meaning. I changed it slightly to preserve its original meaning (hopefully without disrupting correctness). 
`that the order is the same` -> `that the order would be the same`
(maybe it just a difference in my mother tongue)

File: source/plugin/project/version-numbers.rst (hunk 3 above)
Comment (ST-DDT) [reply source/plugin/project/version-numbers.rst:32]: Fixed

**Pull request merged**